lib/src/view/analysis/tree_view.dart

```diff
@@ -162,124 +131,654 @@ class _InlineTreeViewState extends ConsumerState<AnalysisTreeView> {
+          if (widget.params.currentPath.isEmpty)
+            SizedBox.shrink(key: widget.params.currentMoveKey),
+
+          if (widget.params.shouldShowComments &&
+              widget.rootComments?.any((c) => c.text?.isNotEmpty == true) ==
+                  true)
+            Text.rich(
+              TextSpan(
+                children:
+                    _comments(widget.rootComments!, textStyle: _baseTextStyle),
+              ),
+            ),
+          ...treeParts.map((part) => part.lines).flattened,
+        ],
+      ),
+    );
+  }
+}
+
+List<InlineSpan> _buildInlineSideLine({
+  required ViewBranch firstNode,
+  required ViewNode parent,
+  required UciPath initialPath,
+  required TextStyle textStyle,
+  required bool followsComment,
+  required _PgnTreeViewParams params,
+}) {
+  textStyle = textStyle.copyWith(
+    fontSize: textStyle.fontSize != null ? textStyle.fontSize! - 2.0 : null,
+  );
+
+  final sidelineNodes = [firstNode, ...firstNode.mainline];
+
+  var path = initialPath;
+  return [
+    if (followsComment) const WidgetSpan(child: SizedBox(width: 4.0)),
+    ...sidelineNodes.mapIndexedAndLast(
+      (i, node, last) {
+        final pathToNode = path;
+        path = path + node.id;
+
+        return [
+          if (i == 0) ...[
+            if (followsComment) const WidgetSpan(child: SizedBox(width: 4.0)),
+            TextSpan(
+              text: '(',
+              style: textStyle,
+            ),
+          ],
+          ..._moveWithComment(
+            node,
+            parent: i == 0 ? parent : sidelineNodes[i - 1],
+            lineInfo: (
+              type: _LineType.inlineSideline,
+              startLine: i == 0 || sidelineNodes[i - 1].hasTextComment
+            ),
+            pathToNode: pathToNode,
+            textStyle: textStyle,
+            params: params,
+          ),
+          if (last)
+            TextSpan(
+              text: ')',
+              style: textStyle,
+            ),
+        ];
+      },
+    ).flattened,
+    const WidgetSpan(child: SizedBox(width: 4.0)),
+  ];
+}
+
+const _baseTextStyle = TextStyle(
+  fontSize: 16.0,
+  height: 1.5,
+);
+
+/// The different types of lines (move sequences) that are displayed in the tree view.
+enum _LineType {
+  /// (A part of) the game's main line.
+  mainline,
+
+  /// A sideline branching off the main line or a parent sideline.
+  /// Each sideline is rendered on a new line and indented.
+  sideline,
+
+  /// A short sideline without any branching, displayed in parantheses inline with it's parent line.
+  inlineSideline,
+}
+
+/// Metadata about a move's role in the tree view.
+typedef _LineInfo = ({_LineType type, bool startLine});
+
+List<InlineSpan> _moveWithComment(
+  ViewBranch branch, {
+  required ViewNode parent,
+  required TextStyle textStyle,
+  required _LineInfo lineInfo,
+  required UciPath pathToNode,
+  required _PgnTreeViewParams params,
+  GlobalKey? moveKey,
```

Comment:
Not sure to understand what this `moveKey` is? Is it the current move key? If so it'd be better named `currentMoveKey`.

Reply:
Well, it's only the `currentMoveKey` if this is the current move :D 

but I could simply make this a bool and let the InlineMove assign the key by itself instead, that would probably be more readable

Reply:
Oh wait, I get confused as well, yeah this definitely needs a comment. This is `firstMoveKey` if it's the start of the sideline (so that we can draw the indent guides) and null otherwise